hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/DatanodeSchemaThreeDBDefinition.java

```diff
@@ -92,14 +93,25 @@ public class DatanodeSchemaThreeDBDefinition
           DeletedBlocksTransaction.class,
           Proto2Codec.get(DeletedBlocksTransaction.class));
 
+  public static final DBColumnFamilyDefinition<String, FinalizeBlockList>
+      FINALIZE_BLOCKS =
+      new DBColumnFamilyDefinition<>(
+          "finalize_blocks",
+          String.class,
+          FixedLengthStringCodec.get(),
+          FinalizeBlockList.class,
+          FinalizeBlockList.getCodec());
+
```

Comment:
Please don't use list as value here.  The list will have performance impact. For example, if there is 99 blocks already, then the 100th block comes, we need to persist the whole list with 100 elements in to RocksDB, the list introduces unnecessary bytes to write.

Reply:
I have changed storage as <ContainerId | LocalId, BlockData>. Since DatanodeTable [iterator](https://github.com/apache/ozone/blob/be2e19948b6c8fb3f3b9a0c69ee6dfd88967f840/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/DatanodeTable.java#L80) doesn't support. I store value as BlockData and currently used [keyValueBlockIterator](https://github.com/apache/ozone/blob/be2e19948b6c8fb3f3b9a0c69ee6dfd88967f840/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/AbstractDatanodeStore.java#L287) which can iterate through blocks. Is this fine to store or do you have any other suggestion?

Reply:
How about implement a new implementation of BlockIterator? Since only the local ID is useful, then the value can be just the local ID.

Reply:
Added new implementation of BlockIterator and store value as localId only.